pkg/chezmoi/sourcestate.go

```diff
@@ -2179,6 +2181,29 @@ func (e *External) OriginString() string {
 	return e.URL + " defined in " + e.sourceAbsPath.String()
 }
 
+func NewConfiguredTemplate(
+	name string,
+	data []byte,
+	funcs template.FuncMap,
+	options ...TemplateOptions,
+) (*template.Template, error) {
+	var o TemplateOptions
+
+	if len(options) > 0 {
+		o = options[0]
+	} else {
+		o = TemplateOptions{}
+	}
```

Comment:
This will silently ignore any `TemplateOptions`after the first one. Instead, I'd suggest that the caller passes the address of an existing struct if they don't want a zero-valued `TemplateOptions`, or `nil` if they do:

```go
func NewConfiguredTemplate(
	name string,
	data []byte,
	funcs template.FuncMap,
	options *TemplateOptions,
) (*template.Template, error) {
	var o TemplateOptions
	if options != nil {
		o = *options
	}
	// ...
}
```

Reply:
Yeah. I was trying to avoid the explicit `nil` in this case.